### ISSUE TITLE
[Bug]: `.js` in package name fails the build (e.g. `highlight.js`)

### ISSUE THREAD

**Issue by markmssd** [web-infra-dev/rspack#6785]
### System Info

  System:
    OS: macOS 14.5
    CPU: (10) arm64 Apple M1 Max
    Memory: 2.63 GB / 32.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 20.14.0 - ~/.nvm/versions/node/v20.14.0/bin/node
    Yarn: 1.22.22 - ~/.nvm/versions/node/v20.14.0/bin/yarn
    npm: 10.7.0 - ~/.nvm/versions/node/v20.14.0/bin/npm
    pnpm: 7.28.0 - ~/Library/pnpm/pnpm
    bun: 1.1.8 - ~/.bun/bin/bun
  Browsers:
    Chrome: 125.0.6422.142
    Edge: 125.0.2535.92
    Safari: 17.5
  npmPackages:
    @rspack/cli: ^0.7.0 => 0.7.2 
    @rspack/core: ^0.7.0 => 0.7.2 

### Details

I was trying to use [react-syntax-highlighter](https://github.com/react-syntax-highlighter/react-syntax-highlighter) or [react-highlight](https://github.com/akiran/react-highlight), but the build was failing. After digging into it, it seems it's because both of them use [highlight.js](https://www.npmjs.com/package/highlight.js) underneath.

Seems that because the package name has the extension `.js` in it, RSPack doesn't handle it properly and crashes with:

<img width="845" alt="Screenshot 2024-06-12 at 3 01 03 PM" src="https://github.com/web-infra-dev/rspack/assets/5899024/19a28541-3bfc-4afb-8342-92e6194d38dd">

### Reproduce link

https://github.com/markmssd/rspack-repro/tree/highlight.js

### Reproduce Steps

```
git clone https://github.com/markmssd/rspack-repro/tree/highlight.js
npm install
npm run dev:rspack
```

This will fail with the above error, due to this import: https://github.com/markmssd/rspack-repro/blob/highlight.js/src/render.js#L1. If you simply comment this line, it will work fine.

**Comment by hulin32** [comment]
@markmssd seems this issue only happened in 0.7.2, you can use 0.7.1 first to bypass it at this moment

**Comment by markmssd** [comment]
Hey you're right, it works with 0.7.1. Unfortunately we can't use that version neither, because it had another issue which is fixed in 0.7.2 :P 

I bisected it to understand where exactly the issue got introduced. In `0.7.2-canary-c0c0fd3-20240607004432`, it was working, then in `0.7.2-canary-3dcf0a9-20240607151650` it broke. Does that mean that https://github.com/web-infra-dev/rspack/commit/3dcf0a9 is the faulty commit? 🤔

**Comment by LingyuCoder** [comment]
This is a known issue and we are fixing it. It is caused by too much string operations in `highlight.js`, which led to rust stack overflow during compile time evaluation.
